### ISSUE TITLE
App Engine Pipeline not aborting all the way to root

### ISSUE THREAD

**Issue by NickFranceschina** [GoogleCloudPlatform/appengine-pipelines#56]
I recently updated to the latest version of AppEngine Pipelines Python library and I'm seeing a problem where a pipeline abortion is not bubbling all the way to to the root. This used to work fine... can anyone shed light on it?   I don't know what other info to provide...

![screen shot 2015-09-01 at 2 40 33 pm](https://cloud.githubusercontent.com/assets/419831/9613266/5b4059fc-50b8-11e5-9fe2-da415bbd16f0.png)

**Comment by NickFranceschina** [comment]
hmmmmm... well I reverted back to previous version of pipeline and the same thing happens... so it must be something I'm doing wrong...

**Comment by NickFranceschina** [comment]
well I have determined it has something to do with an "extremely esoteric problem" (see link below) around mixing sync and async calls within a tasklet.  I'm not sure how it was swallowing up the errors and keeping it from aborting all the way to the root... but it did.  I refactored the code so that there was no mixing, and now things are working fine.   

https://groups.google.com/forum/#!topic/google-appengine/t2AZkfmoUlk
